### ISSUE TITLE
ExpressionBodySyntax false positive

### ISSUE THREAD

**Issue by Mygod** [arturbosch/detekt#842]
For example, here: https://github.com/shadowsocks/shadowsocks-android/blob/a7d7746195d687619899ac7b73fed7791eea9492/mobile/src/main/java/com/github/shadowsocks/ProfilesFragment.kt#L364-L376

Running 563e6a3 + #840.

**Comment by schalkms** [comment]
I just saw this issue. You can refactor that code to use the expression body syntax. Thus, the warning gets triggered.
```Kotlin
override fun onMenuItemClick(item: MenuItem) = when (item.itemId) {
  // ...
}
```

**Comment by Mygod** [comment]
I think this is different from multiline expressions. Some function can be short and still impossible to use expression body syntax, for example `return (nullable ?: return Stub).generateStub()`.

**Comment by arturbosch** [comment]
Thanks for pointing this out. I've fixed it for the next release.
